### PR TITLE
FIX: explicit keyword argument for Horizon

### PR DESCRIPTION
**Pull request by skoudoro**
The last release of FURY enforces keyword-only arguments. 

This new feature generates a ton of warning in our codebase so this PR just fix this issue and make everything more explicit.

viz CI's should be back to normal after this PR

### CODE REVIEW

**Comment by skoudoro** [comment]
This PR is ready to go. If someone could have a look @jhlegarreta or @arokem ?

**Review by jhlegarreta — APPROVED** [review]
Test failures are unrelated.

Thanks for doing this.

File: dipy/viz/tests/test_apps.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ def test_horizon(rng):
     streamlines.append(s1)
     streamlines.append(s2)
     streamlines.append(s3)
+    streamlines.shrink_data()
```
Comment (jhlegarreta) [inline dipy/viz/tests/test_apps.py:108]: IMO Deserves a separate commit, since it looks like this is the change that will fix #3231, the rest of changes being STYLE changes.

Comment (skoudoro) [reply dipy/viz/tests/test_apps.py:108]: done in 3bbdeccbf3010b0c9b44f547e6c3fa5ac8d9b8f4

**Comment by skoudoro** [comment]
When CI finishes, I will go ahed and merge this if no objection

File: dipy/viz/tests/test_apps.py (hunk 1 above)
Comment (skoudoro) [reply dipy/viz/tests/test_apps.py:108]: > the rest of changes being STYLE changes.

It was not style since both were fixing warnings generated by the new FURY release and keyword only arguments rules

Comment (skoudoro) [reply dipy/viz/tests/test_apps.py:108]: see here https://github.com/dipy/dipy/actions/runs/10199389702/job/28216359136#step:11:4208

Comment (jhlegarreta) [reply dipy/viz/tests/test_apps.py:108]: > It was not style since both were fixing warnings generated by the new FURY release and keyword only arguments rules

OK for FIX then :+1:. Thanks for having created a separate commit for the issue with the array.

**Comment by jhlegarreta** [comment]
> When CI finishes, I will go ahed and merge this if no objection

Yes, please. Thanks.

**Comment by skoudoro** [comment]
Failed tests not related. All good, Merging
